components/profile_popover/index.js

```diff
@@ -39,15 +39,19 @@ function mapStateToProps(state, ownProps) {
     }
 
     const selectedPost = getSelectedPost(state);
-    let channelId;
-    if (selectedPost.exists === false) {
-        const currentChannel = getCurrentChannel(state) || {};
-        channelId = currentChannel.id;
-    } else {
-        channelId = selectedPost.channel_id;
+
+    let channelId = ownProps.channelId;
+
+    if (!channelId) {
+        if (selectedPost.exists === false) {
+            const currentChannel = getCurrentChannel(state) || {};
+            channelId = currentChannel.id;
+        } else {
+            channelId = selectedPost.channel_id;
```

Comment:
can we have a positive condition instead? it is usually easier to read for people
```suggestion
        if (selectedPost.exists) {
            channelId = selectedPost.channel_id;
        } else {
            channelId = getCurrentChannel(state)?.id;
```

Reply:
Would've used this, but N/A anymore per above